### PR TITLE
[vcpkg baseline][ideviceinstaller] Fixing error LNK2005 when argtable2 installed before ideviceinstaller

### PR DESCRIPTION
**Pull request by Cheney-W**
**Describe the pull request**

- #### What does your PR fix?
  ideviceinstaller:x64-windows-static install failed, this error is currently blocking some PRs.
  This issue could be fixed by using getopt as a dependence of argtable2.

  Below is the detailed info of this error:
``` 
  Failure:
  getopt.lib(getopt.c.obj) : error LNK2005: optind already defined in argtable2.lib(getopt.c.obj)
  getopt.lib(getopt.c.obj) : error LNK2005: opterr already defined in argtable2.lib(getopt.c.obj)
  getopt.lib(getopt.c.obj) : error LNK2005: optopt already defined in argtable2.lib(getopt.c.obj)
```
  This issue only occurs when argtable2 installed before ideviceinstaller.
  getopt is a dependence of ideviceinstaller, argtable2 is not, but getopt and argtable2 have same symbols:
```
  src\CMakeFiles\argtable2.dir\getopt.c.obj    opterr
  src\CMakeFiles\argtable2.dir\getopt.c.obj    optind
  src\CMakeFiles\argtable2.dir\getopt.c.obj    optopt

  CMakeFiles\getopt.dir\getopt.c.obj    opterr
  CMakeFiles\getopt.dir\getopt.c.obj    optind
  CMakeFiles\getopt.dir\getopt.c.obj    optopt
```
  Since ideviceinstaller installed by vcpkg_install_msbuild() and use USE_VCPKG_INTEGRATION parameter,  it will link all of the .lib file under vcpkg/installed\x64-windows-static\lib, then the error LNK2005 occurs.

### CODE REVIEW

**Comment by Neumann-A** [comment]
> argtable2 is not, but getopt and argtable2 have same symbols:

So argtable is missing a dependency on getopt ?

**Comment by Neumann-A** [comment]
Yeah after checking argtable2. 
https://github.com/jonathanmarvens/argtable2/blob/b47cd9b9a13405f4d4656a89bde2c00896c98d82/CMakeLists.txt#L10
is definitely missing a dependency on getopt.

File: ports/ideviceinstaller/vcpkg.json [hunk 1]
```diff
@@ -1,11 +1,11 @@
 {
   "name": "ideviceinstaller",
   "version": "1.1.2.23",
-  "port-version": 5,
+  "port-version": 6,
   "description": "Manage apps of iOS devices",
   "homepage": "https://libimobiledevice.org/",
   "license": "LGPL-2.1-only",
-  "supports": "windows & !arm64",
+  "supports": "windows & !arm64 & !static",
```
Comment (Neumann-A) [inline ports/ideviceinstaller/vcpkg.json:8]: This is incorrect. argtable2 is missing the dependency on getopt and setting the appropriate option.

Comment (Cheney-W) [reply ports/ideviceinstaller/vcpkg.json:8]: OK, I will follow this way to solve this issue. Thank you!

**Comment by Neumann-A** [comment]
Be aware: You probably need to patch the CMake buildsystem to actually respect that setting.

**Comment by Neumann-A** [comment]
Also the error in `ideviceinstaller` was due to the `msbuild` vcpkg integration wasn't it?

**Comment by Cheney-W** [comment]
It seems that not only `argtable2` himself compiled `getopt`, `gdcm` also exists this situation.

**Comment by cnSchwarzer** [comment]
Still not fixed in last ci run

**Comment by dg0yt** [comment]
Now it is vendored getopt in darknet.lib..

**Comment by Cheney-W** [comment]
Fixing by https://github.com/microsoft/vcpkg/pull/28718.

**Comment by tomghuang** [comment]
@Cheney-W , it has been a long time since you added a patch for the argtable3 port, so I'm not sure if you still remember the details. However, I'm curious why you had to patch argtable3 while this issue is about argtable2? Thanks.

**Comment by Cheney-W** [comment]
The reason for modifying argtable3 was that it could trigger issues with ideviceinstaller at the time. However, I've now found that even after reverting the previous changes, this port no longer affects ideviceinstaller. I suspect this is due to updates in ideviceinstaller.

**Pull request merged**